src/idl_gen_cpp.cpp

```diff
@@ -569,26 +569,46 @@ class CppGenerator : public BaseGenerator {
                             : name;
   }
 
-  const std::string &PtrType(const FieldDef *field) {
-    auto attr = field ? field->attributes.Lookup("cpp_ptr_type") : nullptr;
-    return attr ? attr->constant : parser_.opts.cpp_object_api_pointer_type;
+  const std::string FieldStringLookup(const FieldDef *field,
+                                      const std::string &attrName) {
```

Comment:
Please use the same code style as surrounding code, e.g. `snake_case` not `camelCase`.